### DISCUSSION TITLE
Simplification

### DISCUSSION THREAD

**Discussion by wer-mathurin** [bahricanyesil/flutter-animated-login#30] [Ideas]
One use case is being able to hide the Welcome title or do more fancy stuff.

What do you think of being able to pass Widget instead of text for LoginText where it make sense and/or completely removing LoginText, I will show you later with example what I have in mind.

By default, if no widget it provided use a Text widget with default TextTheme. (exactly like today)
This can also remove the need of passing the TextTheme to the loginTheme.

If you need you style the Welcome here is what I have in mind...

If you need to change the Style

```
LoginTheme(
   welcome: (defaultText, languageOption) => Text(defaultText, style: TextStyle(...))
);
```
If you need to remove the widget and space.

```
LoginTheme(
   welcome: (defaultText, languageOption) => Container()
);
```
By providing the languageOption is simpler to display the appropriate message

```
LoginTheme(
   welcome: (defaultText, languageOption) => Text(MY_LOCALIZED_MESSAGE(languageOption))
);
```
For customizing the TextFormField maybe we can use the standard InputDecoration instead of passing the text in LoginText and style in LoginTheme.

```
LoginTheme(
   login: (decoration, languageOption) => decoration.copyWith(...)
);
```
I think this will remove the need for multiple properties like welcomePadding. Since you can change the padding by providing yours :-)

**Comment by bahricanyesil** [comment]
Hi @wer-mathurin , I couldn't find enough time to analyze and think about it in a detail. However, you pointed out important parts, thank you! Firstly, this is what I have understood and came into my mind about the Text widgets:
 
As you said we can change these options and enable users to provide their own custom text widgets (maybe they need a text widget with more parameters). So, I think what you said is sensible. 
However, one problem here is I'm using the widget I created which is `BaseText` for text widgets across the app. It allows fitting texts into the available space and scaling up/down accordingly.

Finally, what I'm thinking to implement is as follows:
- Allow users to pass their own customized text widgets
- While doing this, also make public the `BaseText` widget for enabling users to use them. 
- Now, parameters of `BaseText` are limited. I will add all possible text widget properties as parameters.

Additionally, about the removing login texts and theme part:
- What about the users who don't want to care about the stuff other than the text itself? If someone just wants to change the text string instead of the complete widget/design and protect the current package implementation. how can we enable this? Basically, how can we both enable users to pass their own widgets and also pass single parameters by using the default widget?

**Comment by wer-mathurin** [comment]
> What about the users who don't want to care about the stuff other than the text itself? If someone just wants to change the text string instead of the complete widget/design and protect the current package implementation. how can we enable this? Basically, how can we both enable users to pass their own widgets and also pass single parameters by using the default widget?

By default, the plugin can be use with nothing to do. Let say your BaseText expose some API.

I include the Type for clarity
```
LoginTheme(
   welcome: (String defaultText, LanguageOption languageOption, BaseText child) => child..text('MY_TEXT')
);
```
